app/models/cms/section.rb

```diff
@@ -204,6 +204,12 @@ def not_own_child
     end
   end
 
+  def parent_same_provider
+    return if parent&.provider == provider
```

Comment:
If the `CMS::Section` is not `root?`, shouldn't there always be a `parent`? Maybe we can remove the safe navigator.

Reply:
If it remove the safe operator, this test fails: [test/unit/cms/section_test.rb#L11-L15](https://github.com/3scale/porta/blob/master/test/unit/cms/section_test.rb#L11-L15)

Reply:
Ah OK... It fails because the test object is a new record and `parent_id` is null. For that case, [this validation](https://github.com/3scale/porta/blob/master/app/models/cms/section.rb#L24) will fire and we'll get a `parent_id can't be blank`.